numpy/core/src/umath/umath_tests.c.src

```diff
@@ -155,6 +155,21 @@ static void
     npy_intp ib2_n = is2_n*dn;
     npy_intp ib2_p = is2_p*dp;
     npy_intp ob_p  = os_p *dp;
+    if (dn == 0) {
+        /* No operand, need to zero the output */
+        BEGIN_OUTER_LOOP_3
+            char *op=args[2];
+            for (m = 0; m < dm; m++) {
+                for (p = 0; p < dp; p++) {
+                    *(@typ@ *)op = 0;
+                    op  +=  os_p;
+                }
+                op  -=  ob_p;
+            op  +=  os_m;
+            }
+        END_OUTER_LOOP
+        return;
+    }
```

Comment:
Wouldn't a better fix be to just reassign the pointers, and run the below code?

Something like

```python
if (dn == 0) {
     @typ@ ip_val = 0;
     @typ@ ip2_val = 0;
      ip1 = ip2 = &ip_val;
      // set strides to 0
      dn = 1
}
```